judge/judgedaemon.main.php

```diff
@@ -16,6 +16,72 @@
 $endpoints = [];
 $domjudge_config = [];
 
+function dj_getopt(string $short_options, array $long_options = []): array
+{
+    global $argv;
+    define('GETOPT_REGEX', "/^([a-zA-Z0-9]:{0,2})*$/");
+    $options = getopt($short_options, $long_options);
+    if (preg_match(GETOPT_REGEX, $short_options) !== 1) {
```

Comment:
You probably want to do the check in line 24 before the `getopt` call in line 23.

Reply:
Done.